pitrery

```diff
@@ -2265,11 +2276,8 @@ case $action in
 			# find the latest backup
 			for t in "${list[@]}"; do
 				# get the timestamp of the end of the backup
-				if [ "$backup_local" = "yes" ]; then
-					backup_timestamp=$(< "$t")
-				else
-					backup_timestamp=$(ssh -n -- "$backup_ssh_target" "cat -- $(qw "$t")")
-				fi
+				get_backup_timestamp $t
+				backup_timestamp="$ret_backup_timestamp"
```

Comment:
Great sorry for the delayed reply.  using a global variable is not the best way to return value. 

Bash isn't great for writing function and return value but there are at least two ways to do that.

1. The simplest one - using echo
```
get_backup_timestamp(){
<snip>
echo "${my_result}"
}
```
Then you can call this function that way :
```
ret="$(get_backup_timestamp)"
``` 
A fully functionnal example : 
```
#!/bin/bash
function get_backup_timestamp(){
  echo "toto"
}

ret="$(get_backup_timestamp)"
echo "$ret"
```
2. Using declare with namerefs (only available since  bash v4.3-alpha or later) :
```
#!/bin/bash
func_tes(){
  declare -n ret=$1
  ret="truc"
}

func_tes my_var
echo "${my_var}"
```
https://linuxhint.com/bash_declare_command/
https://stackoverflow.com/questions/3236871/how-to-return-a-string-value-from-a-bash-function

Personally I prefer the first solution

Reply:
I tried to implement (1) in the PR.